src/query/builder.js

```diff
@@ -751,11 +755,23 @@ assign(Builder.prototype, {
     let amt = parseInt(amount, 10);
     if (isNaN(amt)) amt = 1;
     this._method = 'counter';
-    this._single.counter = {
-      column,
-      amount: amt,
-      symbol: (symbol || '+')
-    };
+    let counter = this._single.counter || {};
+
+    if(!counter[column]) {
+      counter[column] = {
+        amount: amt,
+        symbol: (symbol || '+')
+      };
+    } else {
+      //Counter for column already exists. Add/subtract the current column amount by the new amount
+      if(symbol === '-') {
+        counter[column].amount -= amount;
```

Comment:
To me actually would make more sense that amount is always overridden / failure would be thrown if one tries to set  increment / decrement twice for the same field.

Using builder to calculate values back and forth sounds inefficient and feels to be something that builder shouldn't doing.  Just overriding / throw when same field is set twice is simpler / more robust to implement and easier to test too.

Anyways just my 2-cents, how I feel that feature should work 👍 I don't have any strong opinion about it.